dialect/sql/schema/writer.go

```diff
@@ -229,6 +231,60 @@ func (w *WriteDriver) formatArg(v any) (string, error) {
 	}
 }
 
+// trimReturning trims any RETURNING suffix from INSERT/UPDATE queries.
+// Note, that the output may be incorrect or unsafe SQL and require manual changes.
+func trimReturning(query []byte) []byte {
+	var returning = []byte("returning")
+	var (
+		b          bytes.Buffer
+		skipQuoted = func(query []byte, index int) ([]byte, int) {
+			for j := index + 1; j < len(query); j++ {
+				switch query[j] {
+				case '\\':
+					j++
+				case query[index]:
+					return query[index : j+1], j
+				}
+			}
+			// Unexpected EOS.
+			return query, -1
+		}
+	)
+loop:
+	for i := 0; i < len(query); i++ {
+		switch query[i] {
+		case '\'', '"', '`': // string or identifier
+			s, skip := skipQuoted(query, i)
+			if skip == -1 {
+				return query
+			}
+			b.Write(s)
+			i = skip
+			continue
+		case ' ', 'r', 'R': // start of RETURNING?
+			if bytes.HasPrefix(bytes.ToLower(bytes.TrimSpace(query[i:])), returning) {
```

Comment:
```suggestion
		switch r := query[i]; {
		case r == '\'', r == '"', r == '`': // string or identifier
			s, skip := skipQuoted(query, i)
			if skip == -1 {
				return query
			}
			b.Write(s)
			i = skip
			continue
		case bytes.HasPrefix(query[i:], returingL), bytes.HasPrefix(query[i:], returningU): // start of RETURNING?
```
More efficient and cleaner, IMO. Just call TrimRight(query, " ") at the end.